server/middleware/current_user.go

```diff
@@ -0,0 +1,64 @@
+package middleware
+
+import (
+	"strings"
+	
+	"github.com/gin-gonic/gin"
+
+	oauth_service "github.com/torinos-io/api/service/oauth_service"
+	user_store "github.com/torinos-io/api/store/user_store"
+	"github.com/torinos-io/api/type/model"
+	"github.com/torinos-io/api/type/system"
+)
+
+const (
+	currentUserContextName = "CurrentUser"
+)
+
+// SetCurrentUser sets current authenticated user from authorization header
+func SetCurrentUser(appContext *system.AppContext) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		h := c.GetHeader("Authorization")
+		
+		if h == "" {
+			c.Next()
+			return
+		}
+		
+		splited := strings.Split(h, "Bearer")
```

Comment:
Well, maybe not.. it leaves undesired space here.

```
"Bearer xxxxxxxxx" --> ["Bearer", " xxxxxxxxx"]
                                   ^
```

Take a look at notifications for example.
https://github.com/wantedly/notifications/blob/ea6a316558508abc529d7e059185c4abd654c03f/server/middleware/current_user.go#L61